CMakeLists.txt

```diff
@@ -86,6 +90,15 @@ target_compile_options(
     PRIVATE ${COMPILER_DIAGNOSTIC_FLAGS}
 )
 
+if(ENABLE_STATIC)
+    # Avoid compiling with fPIC for static builds, this can often lead to
+    # GCC disabling many optimizations e.g. inlining, possibly introducing
+    # an overhead.
+    set_property(TARGET wiredtiger PROPERTY POSITION_INDEPENDENT_CODE OFF)
+else()
+    set_property(TARGET wiredtiger PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
What happens if you build a shared lib with `-fPIC` anyway? Do bad things happen.

Reply:
We will unfortunately generate dreaded linker errors, failing to compile the shared library. They are typically of the type `relocation R_X86_64_PC32 against symbol ...`. This is mainly associated with our large number of`extern` functions found in `src/include/extern.h`. We need to compile with PIC so we can call external symbols (function calls across shared-object boundaries).

Currently we depend on libtool to do all the PIC magic, which is why we don't currently see with `-fPIC` in our compilation output (libtool generates object files `libs/*.lo`file for each object. This managing PIC objects when performing the final linker step of shared library). 
Since we are not using libtool, we need to ensure `-fPIC` is present when compiling a shared library with CMake.